src/site/controller.ts

```diff
@@ -3,44 +3,187 @@
  *
  * This is where we handle basic functions of sites.
  */
-
 import { Request, Response, Router } from 'express';
+import { ObjectId } from 'mongoose';
+import { UserModel } from '../user/model';
+import { checkId, ExpressHandler, HTTPError } from '../utils';
+import { SiteModel } from './model';
+
+const SiteRouter = Router();
+export default SiteRouter;
+
+SiteRouter.get('/', getAllSites);
+SiteRouter.put('/', createSite);
+SiteRouter.delete(
+  '/',
+  async (req, res, next) =>
+    await checkId(res, next, SiteModel, req.body.siteId),
+  closeSite,
+);
 
-const router = Router();
+SiteRouter.get(
+  '/:siteId',
+  async (req, res, next) =>
+    await checkId(res, next, SiteModel, req.params.siteId),
+  getOneSite,
+);
 
-router.get('/', getAllSites);
-router.put('/', createSite);
-router.delete('/', closeSite);
+SiteRouter.get(
+  '/:siteId/members',
+  async (req, res, next) =>
+    await checkId(res, next, SiteModel, req.params.siteId),
+  getSiteMembers,
+);
 
-router.get('/:siteId', getOneSite);
-router.get('/:siteId/members', getSiteMembers);
-router.put('/:siteId/members', assignMemberToSite);
-router.delete('/:siteId/members', removeMemberFromSite);
+SiteRouter.put(
+  '/:siteId/members',
+  async (req, res, next) =>
+    await checkId(res, next, SiteModel, req.params.siteId),
+  async (req, res, next) =>
+    await checkId(res, next, UserModel, req.body.memberId),
+  assignMemberToSite,
+);
 
-function getAllSites(_req: Request, res: Response) {
-  res.sendStatus(501);
+SiteRouter.delete(
+  '/:siteId/members',
+  async (req, res, next) =>
+    await checkId(res, next, SiteModel, req.params.siteId),
+  removeMemberFromSite,
+);
+
+/**
+ * Get a list of all sites
+ */
+async function getAllSites(_req: Request, res: Response) {
+  ExpressHandler(
+    async () => {
+      const sites = await SiteModel.find().lean();
+      const data = { sites };
+      return { data };
+    },
+    res,
+    'getting all sites',
+  );
 }
 
-function createSite(_req: Request, res: Response) {
-  res.sendStatus(501);
+async function createSite(_req: Request, res: Response) {
+  ExpressHandler(
+    async () => {
+      const { name, location, timezone } = _req.body;
+
+      if (!name) {
+        return HTTPError('Missing required arg, name.', 400);
+      }
+
+      if (!location) {
+        return HTTPError('Missing required arg, location.', 400);
+      }
+
+      if (!timezone) {
+        return HTTPError('Missing required arg, timezone.', 400);
```

Comment:
Is there a library we could use to check for valid timezone strings ?

Reply:
I was going to use Moment, but it is deprecated now. Instead changed to using the Intl object that is compatible with our versions of node. MR is out for it here: https://github.com/conturo-prototyping/pack-ship-server/pull/133